### ISSUE TITLE
Pixel font size 

### ISSUE THREAD

**Issue by codecat** [SFML/SFML#1456]
## Subject of the issue

I'm trying to load a pixel font (specifically, artwiz-aqui) and setting the character size to 12, but I get the following error:

```
Failed to set bitmap font size to 12
Available sizes are: 12
```

I think this is because of this line:

https://github.com/SFML/SFML/blob/949e7aecae1101348308433ab3758f85ca5e2b9e/src/SFML/Graphics/Font.cpp#L768

According to [this page](https://www.freetype.org/freetype2/docs/tutorial/step1.html), "a value of 0 for one of the dimensions means ‘same as the other’". So, would it be possible that it's not the height that's not available, but the width?

## Your environment

* Windows 7 Ultimate x64
* SFML 2.5.0
* Visual Studio 15.7.3

## Steps to reproduce

Get `aqui.pcf` from here: https://sourceforge.net/projects/artwizaleczapka/

```cpp
#include <SFML/Graphics.hpp>

int main()
{
	sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
	window.setFramerateLimit(60);

	sf::Font font;
	font.loadFromFile("aqui.pcf");

	sf::Text text;
	text.setFont(font);
	text.setCharacterSize(12);
	text.setString("Hello, world");
	text.setFillColor(sf::Color::White);

	while (window.isOpen())
	{
		sf::Event event;
		while (window.pollEvent(event))
		{
			if (event.type == sf::Event::Closed)
				window.close();
		}

		window.clear();
		window.draw(text);
		window.display();
	}
}
```

## Expected behavior

The font should be 12px high.

## Actual behavior

Strange error.

**Comment by Qu3tzal** [comment]
Hello,

I looked into your problem and found out that your font is only available in 9x12 size (width x height).

**Comment by LaurentGomila** [comment]
Or we could search into `face->fixed_sizes` for the matching width and use it instead of 0, in case of a bitmap font.

**Comment by MarioLiebisch** [comment]
Sounds like a good idea, just pick the size available that's closest to the one requested.

**Comment by Qu3tzal** [comment]
Something among the lines of 19cf28319b97a7b4941fe1ec8f63028f0a152f4f ?

**Comment by LaurentGomila** [comment]
Yes, that's the idea.

**Comment by eXpl0it3r** [comment]
@Qu3tzal Are you going to create a PR?

**Comment by eXpl0it3r** [comment]
The font never had a character size of 12, but the error message returned the wrong value.

Fixed in a5d45b910f380560a928a5670d41e427a951260e
